compiler/src/iree/compiler/Reducer/Framework/WorkItem.h

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#ifndef IREE_COMPILER_REDUCER_WORK_ITEM_H
+#define IREE_COMPILER_REDUCER_WORK_ITEM_H
+
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinOps.h"
+
+namespace mlir {
+namespace iree_compiler {
+
+class WorkItem {
+public:
+  WorkItem(ModuleOp root) : root(root) {}
+
+  const ModuleOp getModule() { return root; }
+  const OpBuilder getBuilder() { return OpBuilder(root); }
+
+  void replaceModule(ModuleOp newModule) {
```

Comment:
I would usually expect to see ownership conveyed here via `mlir::OwningOpReference<ModuleOp>` (unique_ptr semantics) or equiv. Since looks like would be a non-trivial refactor, I'm ok if you add a TODO to switch to that and get it in a followup.

Reply:
Added a TODO for now